script.js

```diff
@@ -1,12 +1,73 @@
 //You can edit ALL of the code here
+
+const search = document.getElementById("search");
+const totalEpisodes = document.getElementById("total-episodes");
+const matchedEpisodes = document.getElementById("matching-ep")
+const allEpisodes =  getAllEpisodes();
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+    displayEpisodes(allEpisodes);
+    selectEpisodes(allEpisodes)
+}
+
+
+//display all episodes
+function displayEpisodes(episodes){
+  const container = document.querySelector(".container");
+  container.innerHTML = "";
+  totalEpisodes.textContent = allEpisodes.length; 
+
+  for(let episode of episodes){
+    const article = document.createElement("article");
+  
+    const title = document.createElement("h2");
+    let formattedEpNumber = ((episode.number.toString()).padStart(2,"0"))// format number to two characters string
+    title.textContent = `${episode.name} - S0${episode.season}E${formattedEpNumber}`;
+    article.appendChild(title);
+    
+    const image = document.createElement("img");
+    image.src = episode.image.medium;
+    article.appendChild(image);
+
+    const p = document.createElement("p");
+    p.innerHTML = `${episode.summary}`;
+    article.appendChild(p);
+
+    container.appendChild(article);
+  }
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+//search episodes
+search.addEventListener('keyup', e => {
+  const matchingValue = e.target.value.toUpperCase();
+  const matchingEpisodes = allEpisodes.filter((episode)=>{
+       return episode.name.toUpperCase().includes(matchingValue) || episode.summary.toUpperCase().includes(matchingValue);
+    })
+  displayEpisodes(matchingEpisodes);
+  matchedEpisodes.innerText = matchingEpisodes.length;
```

Comment:
Ref my general comment about displaying the right number of episodes, have a think about where you could move this to, so that it works for the dropdown selector too